src/micro_http/src/common/headers.rs

```diff
@@ -286,6 +298,72 @@ impl Headers {
     }
 }
 
+/// Wrapper over supported AcceptEncoding.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct Encoding {}
+
+impl Encoding {
+    /// Parses a byte slice into a Encoding structure for a HTTP request. Encoding
+    /// must be ASCII, so also UTF-8 valid.
+    ///
+    /// # Errors
+    /// The function returns `InvalidRequest` when the byte stream is empty, `InvalidValue` when the
+    /// byte stream contains a logical error, `UnsupportedValue` when the byte stream contains a
+    /// value that is not supported yet and `InvalidUtf8String` when the byte stream contains
+    /// invalid characters.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use micro_http::Encoding;
+    ///
+    /// assert!(Encoding::try_from(b"deflate").is_ok());
+    /// assert!(Encoding::try_from(b"indentity;q=0").is_err());
+    /// ```
+    pub fn try_from(bytes: &[u8]) -> Result<(), RequestError> {
+        if bytes.is_empty() {
+            return Err(RequestError::InvalidRequest);
+        }
+        match std::str::from_utf8(bytes) {
+            Ok(headers_str) => {
+                let q_value_identity = headers_str.contains("identity");
+                let entry = headers_str.split(',').collect::<Vec<&str>>();
+                let compression_types = ["br", "compress", "deflate", "gzip", "identity", "*"];
+                for encoding in entry {
+                    let compression = encoding.split(';').collect::<Vec<&str>>()[0].trim();
+                    if !compression_types.contains(&compression) {
+                        return Err(RequestError::HeaderError(
+                            HttpHeaderError::UnsupportedValue(
+                                "Accept-Encoding".to_string(),
+                                encoding.to_string(),
+                            ),
+                        ));
+                    }
+                    match encoding.trim() {
+                        "identity;q=0" => {
+                            Err(RequestError::HeaderError(HttpHeaderError::InvalidValue(
+                                "Accept-Encoding".to_string(),
+                                encoding.to_string(),
+                            )))
+                        }
+                        "*;q=0" if !q_value_identity => {
+                            Err(RequestError::HeaderError(HttpHeaderError::InvalidValue(
+                                "Accept-Encoding".to_string(),
+                                encoding.to_string(),
+                            )))
+                        }
+                        _accepted_encoding => Ok(()),
```

Comment:
I would drop the `_accepted_encoding` name as it is not only not used, but misleading; it implies that we accept multiple encodings, when we only do `identity`.
```suggestion
                        _ => Ok(()),
```